examples/etl_example.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+"""
+Example of using Celery canvas features to build an ETL pipeline.
+This example demonstrates how to chain and group tasks for data processing.
+
+This ETL pipeline example shows how to use Celery's powerful canvas features
+to build a data processing workflow that:
+1. Extracts data from multiple sources in parallel
+2. Merges the extracted data
+3. Transforms the data with computed fields and filtering
+4. Loads the results into a JSON file
+
+Key Celery Features Demonstrated:
+- Parallel task execution using `group`
+- Task chaining using the `|` operator
+- Result passing between tasks
+- Error handling and logging
+
+Pipeline Flow:
+    [extract(source_a), extract(source_b), extract(source_c)] (in parallel)
+                           ↓
+                     merge_data
+                           ↓
+                      transform
+                           ↓
+                        load
+
+To run this example:
+
+1. Start Redis (broker and result backend):
+   ```bash
+   redis-server
+   ```
+
+2. Start a Celery worker:
+   ```bash
+   celery -A examples.etl_example worker --loglevel=INFO
+   ```
+
+3. Run the example:
+   ```bash
+   python examples/etl_example.py
+   ```
+
+The pipeline will extract sample data from three sources in parallel,
+merge the results, apply transformations (adding computed fields and filtering),
+and save the final results to a timestamped JSON file.
+"""
```

Comment:
I asked the agent to add this description, which it first presented to me, as comment in the example.